salt/modules/nexus.py

```diff
@@ -0,0 +1,501 @@
+
+    log.debug('snapshot_version_metadata_xml=%s', snapshot_version_metadata_xml)
+    return snapshot_version_metadata_xml
+
+
+def _get_snapshot_version_metadata(nexus_url, repository, group_id, artifact_id, version, headers):
+    metadata_xml = _get_snapshot_version_metadata_xml(nexus_url=nexus_url, repository=repository, group_id=group_id, artifact_id=artifact_id, version=version, headers=headers)
+    metadata = ET.fromstring(metadata_xml)
+
+    assert group_id == metadata.find('groupId').text
+    assert artifact_id == metadata.find('artifactId').text
+    assert version == metadata.find('version').text
+
+    snapshot_versions = metadata.find('versioning').find('snapshotVersions')
+    extension_version_dict = {}
+    for snapshot_version in snapshot_versions:
+        extension = snapshot_version.find('extension').text
+        value = snapshot_version.find('value').text
+        extension_version_dict[extension] = value
+        if snapshot_version.find('classifier') is not None:
+            classifier = snapshot_version.find('classifier').text
+            extension_version_dict[classifier] = value
+
+    return {
+        'snapshot_versions': extension_version_dict
+    }
+
+
+def __save_artifact(artifact_url, target_file, headers):
+    log.debug("__save_artifact(%s, %s)", artifact_url, target_file)
+    result = {
+        'status': False,
+        'changes': {},
+        'comment': ''
+    }
+
+    if os.path.isfile(target_file):
+        log.debug("File {0} already exists, checking checksum...".format(target_file))
+        checksum_url = artifact_url + ".sha1"
+
+        checksum_success, artifact_sum, checksum_comment = __download(checksum_url, headers)
+        if checksum_success:
+            log.debug("Downloaded SHA1 SUM: %s", artifact_sum)
+            file_sum = __salt__['file.get_hash'](path=target_file, form='sha1')
+            log.debug("Target file (%s) SHA1 SUM: %s", target_file, file_sum)
+
+            if artifact_sum == file_sum:
+                result['status'] = True
+                result['target_file'] = target_file
+                result['comment'] = 'File {0} already exists, checksum matches with nexus.\n' \
+                                    'Checksum URL: {1}'.format(target_file, checksum_url)
+                return result
+            else:
+                result['comment'] = 'File {0} already exists, checksum does not match with nexus!\n'\
+                                    'Checksum URL: {1}'.format(target_file, checksum_url)
+
+        else:
+            result['status'] = False
+            result['comment'] = checksum_comment
+            return result
+
+    log.debug('Downloading: {url} -> {target_file}'.format(url=artifact_url, target_file=target_file))
+
+    try:
+        request = urllib.request.Request(artifact_url, None, headers)
+        f = urllib.request.urlopen(request)
+        with salt.utils.fopen(target_file, "wb") as local_file:
```

Comment:
This has been moved to `salt.utils.files.fopen`. Can you update this reference?

Reply:
Fixed